src/board/system76/addw2/include/board/keymap.h

```diff
@@ -46,4 +46,8 @@
 // Keymap
 extern uint16_t __xdata KEYMAP[KM_LAY][KM_OUT][KM_IN];
 
+// Physical location of the Esc and Fn keys
+#define K_PHYS_ESC KEYMAP[0][7][7]
```

Comment:
Two keys could be set to the same value. I'd recommend defining something like ESC_INPUT and ESCAPE_OUTPUT so it tests i and j against them